sympy/utilities/tests/test_lambdify.py

```diff
@@ -775,11 +774,9 @@ def test_Indexed():
     assert lambdify(a, Sum(a[x, y], (x, 0, 1), (y, 0, 1)))(b) == 10
 
 def test_issue_12173():
-    exp1= lambdify((x,y), uppergamma(x,y),"mpmath")(1,2)
-    exp2= lambdify((x,y), lowergamma(x,y),"mpmath")(1,2)
-    exp1=simplify(exp1)
-    exp2=simplify(exp2)
-    ans1=(uppergamma(1,2)).evalf()
-    ans2=(lowergamma(1,2)).evalf()
+    exp1=N(lambdify((x,y), uppergamma(x,y),"mpmath")(1,2))
+    exp2= N(lambdify((x,y), lowergamma(x,y),"mpmath")(1,2))
+    ans1=N((uppergamma(1,2)))
+    ans2=N((lowergamma(1,2)))
     assert exp1==ans1
```

Comment:
If the automatic comparisons work as expected, this line could be
```
    assert exp1 == uppergamma(1, 2).evalf()
```
The difference between N and evalf is that N will begin by sympifying its argument. In this case we know that `uppergamma(1, 2)` is a SymPy object. Hence it is preferable to use evalf.

(The mpf object exp1 on the left hand side can be directly compared with the SymPy Float on the right hand side. Hence N is not necessary.)